src/editor.js

```diff
@@ -401,6 +486,8 @@ class Editor {
       isFullPreview: false,
       previewElement,
       viewportId: viewportIdDefault,
+      isTemplateLightboxDisplayed: false,
+      templates: fetch('/templates.json').then(r => r.json()),
```

Comment:
Let's use `win.fetch` for tests. Let's also assert that the response `200`s correctly:

```
{
  templates: win.fetch('/templates.json').then(response => {
    assert(response.status == 200);
    return response.json();
  }),
}
```

Reply:
If this is just for tests, would we need to change this later on?

Reply:
Nope! `win` is injected so in tests we can just insert a fake `win` with a fake `fetch`

Reply:
(and code stays like it's needed for the tests boundaries, it's just a matter of separating dependencies so they can be [mocked](https://sinonjs.org/releases/latest/mocks/).